### ISSUE TITLE
`%in.preserve.na%`

### ISSUE THREAD

**Issue by gergness** [Crunch-io/rcrunch#447]
Not sure if I'm missing something, or if there's a DB function that does this that we could use, but I frequently find myself rewriting something like:

```r
cats <- categories(ds$var)[!is.na(categories(ds$var))]
ds$var %in% cats
```

to:
```r
cats <- categories(ds$var)[!is.na(categories(ds$var))]
Reduce(`|`, lapply(names(cats), function(cat) ds$var == cat))
```

If there's not already db support, we could do:
``` r
`%in.preserve.na%` <- function(x, table) {
    Reduce(`|`, lapply(table, function(table_item) x == table_item))
}

c(1, 2, 3) %in% c(1, 3)
#> [1]  TRUE FALSE  TRUE
c(1, 2, NA) %in% c(1, 3)
#> [1]  TRUE FALSE FALSE

c(1, 2, 3) %in.preserve.na% c(1, 3)
#> [1]  TRUE FALSE  TRUE
c(1, 2, NA) %in.preserve.na% c(1, 3)
#> [1]  TRUE FALSE    NA
```

<sup>Created on 2020-05-06 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by gergness** [comment]
(I don't love the name)

**Comment by gergness** [comment]
exists in db as `combine_responses` according to JJ

**Comment by gergness** [comment]
closed in #464
